solutions/cses-1621.mdx

```diff
@@ -96,3 +96,84 @@ print(ans)
 </PySection>
 
 </LanguageSection>
+
+## Alternative Solution
+	
+By using an `std::set` in C++ or its equivalents in other languages, we can take advantage of the fact that a set only stores a copy of an element once.
+Sets and Maps are introduced [here](https://usaco.guide/bronze/intro-sets). Furthermore, using (an unordered set)[https://usaco.guide/gold/unordered] will enable an $\mathcal{O}(n)$ complexity, although $\texttt{chash}$ is necessary to avoid a TLE verdict. 
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N\log{N})$
+
+	
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+ 
+int N;
+set<int> S;
+ 
+int main() {
+	cin >> N;
+	for (int i = 0; i < N; i++) {
+		int a; cin >> a;
+		S.insert(a);
+	}
+	cout << S.size() << '\n';
+}
+```
+</CPPSection>
+<JavaSection>
+
+```java
+// Source: Daniel
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.HashSet;
+import java.util.StringTokenizer;
+
+public class DistinctNumbers {
+
+	public static void main(String[] args) throws IOException {
+		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+		int n = Integer.parseInt(br.readLine());
+		StringTokenizer st = new StringTokenizer(br.readLine());
+		HashSet<Integer> set = new HashSet<Integer>();
+		for (int i = 0; i < n; i++) {
+			set.add(Integer.parseInt(st.nextToken()));
+		}
+		System.out.println(set.size());
+	}
+
+}
+```
+
+</JavaSection>
+
+<PySection>
+
+```py
+n = int(input()) # unused
+nums = [int(x) for x in input().split()]
+distinct_nums = set(nums)
+print(len(distinct_nums))
+```
+
+We can do this more efficiently by skipping the creation of the list, and use a
+set comprehension directly:
+
+```py
+n = int(input()) # unused
+distinct_nums = {int(x) for x in input().split()}
+print(len(distinct_nums))
```

Comment:
```suggestion
```py
n = int(input()) # unused
distinct_nums = {int(x) for x in input().split()}
print(len(distinct_nums))
```

just implement the more efficient solution 
its not that much different than the first that it deserves to be an entire new block of code

Reply:
oops i think that the ``` at the end of the python code were removed when I formatted this comment make sure to add them back

Reply:
@SansPapyrus683 😏 i vow to be more nitpicky than you in the future